apps/easypid/src/features/activity/FunkeActivityDetailScreen.tsx

```diff
@@ -38,27 +37,43 @@ export function FunkeActivityDetailScreen() {
           <HeroIcons.ShieldCheckFilled strokeWidth={2} color="$positive-500" size={56} />
         </YStack>
         <YStack gap="$4" px="$4" marginBottom={bottom}>
-          {activity.disclosedPayload ? (
-            <>
-              <Stack gap="$2" ai="center">
-                <Heading textAlign="center" variant="h1">
-                  {activityTitleMap[activity.type]}
-                </Heading>
-                <Paragraph textAlign="center" color="$grey-700">
-                  You have shared this data with {activity.entityName ?? activity.entityHost} on{' '}
-                  {new Date(activity.date).toLocaleDateString('en-US', {
-                    year: 'numeric',
-                    month: 'long',
-                    day: 'numeric',
-                  })}
-                  .
-                </Paragraph>
-              </Stack>
-              <CredentialAttributes disableHeader subject={activity.disclosedPayload} headerStyle="small" />
-            </>
-          ) : (
-            <Paragraph>Disclosed information could not be shown.</Paragraph>
-          )}
+          <Stack gap="$2" ai="center">
+            <Heading textAlign="center" variant="h1">
+              {activityTitleMap[activity.type]}
+            </Heading>
+            <Paragraph textAlign="center" color="$grey-700">
+              You have shared this data with {activity.entityName ?? activity.entityHost} on{' '}
+              {new Date(activity.date).toLocaleDateString('en-US', {
+                year: 'numeric',
+                month: 'long',
+                day: 'numeric',
+              })}
+              .
+            </Paragraph>
+          </Stack>
+          <Stack py="$4" gap="$4">
+            {activity.credentials && activity.credentials.length > 0 ? (
+              activity.credentials?.map((activityCredential) => {
+                const credential = credentials.find((credential) => credential.id.includes(activityCredential.id))
```

Comment:
I've added that when the credential is not found it will still show the card but just with the name 'Unknown credential' and default branding. So now it will still show the requested attrs and disclosed payload when you've deleted the credential.